code/modules/mob/living/carbon/carbon.dm

```diff
@@ -484,55 +486,53 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
-			visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
-							"<span class='notice'>You begin climbing into the ventilation system...</span>")
+	if(!vent_found)
+		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+		return
 
-			if(!do_after(src, 45, target = src))
-				return
+	if(vent_found.parent && (length(vent_found.parent.members) || vent_found.parent.other_atmosmch))
+		visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
+						"<span class='notice'>You begin climbing into the ventilation system...</span>")
 
-			if(has_buckled_mobs())
-				to_chat(src, "<span class='warning'>You can't vent crawl with other creatures on you!</span>")
-				return
+		if(!do_after(src, 4.5 SECONDS, target = src))
+			return
 
-			if(buckled)
-				to_chat(src, "<span class='warning'>You cannot crawl into a vent while buckled to something!</span>")
-				return
+		if(!client)
+			return
 
-			if(!client)
-				return
+		if(!vent_found.can_crawl_through())
+			to_chat(src, "<span class='warning'>You can't vent crawl through that!</span>")
+			return
 
-			if(iscarbon(src) && contents.len && ventcrawlerlocal < 2)//It must have atleast been 1 to get this far
-				for(var/obj/item/I in contents)
-					var/failed = 0
-					if(istype(I, /obj/item/bio_chip))
-						continue
-					if(istype(I, /obj/item/reagent_containers/patch))
-						continue
-					if(I.flags & ABSTRACT)
-						continue
-					else
-						failed++
-
-					if(failed)
-						to_chat(src, "<span class='warning'>You can't crawl around in the ventilation ducts with items!</span>")
-						return
+		if(has_buckled_mobs())
+			to_chat(src, "<span class='warning'>You can't vent crawl with other creatures on you!</span>")
+			return
+
+		if(buckled)
+			to_chat(src, "<span class='warning'>You cannot crawl into a vent while buckled to something!</span>")
+			return
 
-			visible_message("<b>[src] scrambles into the ventilation ducts!</b>", "You climb into the ventilation system.")
-			var/old_loc = loc
-			loc = vent_found
-			Moved(old_loc, get_dir(old_loc, loc), FALSE)
-			add_ventcrawl(vent_found)
+		if(iscarbon(src) && length(contents) && ventcrawlerlocal < VENTCRAWLER_ALWAYS) // If we're here you can only ventcrawl while completely nude
+			for(var/obj/item/I in contents)
+				if(istype(I, /obj/item/bio_chip))
+					continue
+				if(istype(I, /obj/item/reagent_containers/patch))
+					continue
+				if(I.flags & ABSTRACT)
+					continue
 
-	else
-		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+				to_chat(src, "<span class='warning'>You can't crawl around in the ventilation ducts with items!</span>")
+				return
 
+		visible_message("<b>[src] scrambles into the ventilation ducts!</b>", "You climb into the ventilation system.")
+		var/old_loc = loc
```

Comment:
i don't think old_loc is used here? best to remove it

Reply:
Done